### PR TITLE
feat(helm-values): Support for bumpVersion

### LINKED ISSUE #8231: Support for bumpVersion when updating helm-values
**What would you like Renovate to be able to do?**

When the container image tags in Helm value files are updated using the `helm-values` manager, it should also support bumping the chart version.

**Did you already have any implementation ideas?**

I've been attempting this in https://github.com/chgl/renovate/tree/add-bumpversion-support-to-helm-values (Let me know if you'd like a WIP PR). But the main challenge I face is that the `helm-values` manager works on `values.yaml` files, but the `packageFile` containing the version that should be bumped is `Chart.yaml`. I can work around this by opening and reading the `Chart.yaml` file and get as far as updating the file contents with a bumped version (https://github.com/chgl/renovate/blob/add-bumpversion-support-to-helm-values/lib/manager/helm-values/update.spec.ts#L38), but now I am not sure how to correctly write back the updated contents without majorly updating `getUpdatedPackageFiles` (it currently incorrectly overwrites `values.yaml` with the updated contents of `Chart.yaml`). 

I'm definitely interested in working on this.

### PR DESCRIPTION
**Pull request by chgl**
## Changes:

Implements the `bumpPackageVersion` interface for the `helm-values` manager. Because `helm-values`' packageFile is different from the file containing the version to be bumped (the `Chart.yaml`) this required adding support for decoupling the two. For now, only one additional `bumpVersionPackageFile` is supported, if the need arises it would probably be cleanest to provide a list of files that will need to be updated should the version be bumped.

I've tested it in the repo here: https://github.com/chgl/renovate-helm-values-bumpversion-test/pull/5/files and it *seems* to work.

## Context:

Closes #8231

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [x] Unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]
Small types change

File: lib/manager/common.ts [hunk 1]
```diff
@@ -229,6 +229,13 @@ export interface UpdateDependencyConfig<T = Record<string, any>> {
 
 export interface BumpPackageVersionResult {
   bumpedContent: string | null;
+  // describes a file that was changed instead of or in addition to the packageFile
+  bumpedFile?: BumpedPackageFile;
```
[the remark this answers is not in the document]
Comment (rarkins) [reply lib/manager/common.ts:233]: I'm starting to think version bumping could/should be part of "artifacts updating" - but I don't object to continuing this PR as is. Agree with the idea to make it an array.

**Comment by rarkins** [comment]
A potential problem in future (not affecting this functionality I think) is that the `Chart.yaml` could have *already* been updated by Renovate and if we read it from disk in this bumpPackageVersion function then we'll get the pre-updated version and not the updated one. But that's a bigger problem to solve.

File: lib/manager/common.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (chgl) [reply lib/manager/common.ts]: should I create an issue for this? "Investigate using artifacts updating for version bumping" (better wording suggestions welcome ;)

**Review by rarkins — DISMISSED** [review]

**Comment by chgl** [comment]
is there

> A potential problem in future (not affecting this functionality I think) is that the `Chart.yaml` could have _already_ been updated by Renovate and if we read it from disk in this bumpPackageVersion function then we'll get the pre-updated version and not the updated one. But that's a bigger problem to solve.

Agreed! In this case, the helmv3 manager is able to bump the version of Chart.yaml as well, however since `bumpVersion` is a global setting, both this and the helmv3 manager would try to bump the version to the same value, overriding each other but still arriving at the same version.

Is there anything else I could improve or address for this PR?

**Review by rarkins — APPROVED** [review]

**Pull request merged**